tests/31sync/09archived.pl

```diff
@@ -121,6 +121,77 @@
    };
 
 
+test "Previously left rooms don't appear in the leave section of sync",
+   requires => [ local_user_fixture( with_events => 0 ), local_user_fixture( with_events => 0 ),
+                 qw( can_sync ) ],
+
+   bug => 'SYN-589',
+
+   check => sub {
+      my ( $user, $user2 ) = @_;
+
+      my ( $filter_id, $room_id_1, $room_id_2, $next );
+
+      my $filter = {
+         room => { timeline => { limit => 1 }, include_leave => JSON::true }
+      };
+
+      matrix_create_filter( $user, $filter )->then( sub {
+         ( $filter_id ) = @_;
+
+         Future->needs_all(
+            matrix_create_room( $user )->on_done( sub { ( $room_id_1 ) = @_; } ),
+            matrix_create_room( $user )->on_done( sub { ( $room_id_2 ) = @_; } ),
+         );
+      })->then( sub {
+         matrix_join_room( $user2, $room_id_1 );
+      })->then( sub {
+         matrix_join_room( $user2, $room_id_2 );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         $next = $body->{next_batch};
+
+         matrix_leave_room( $user, $room_id_1 );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id, since => $next );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         $next = $body->{next_batch};
```

Comment:
Do you need to assert that you definitely did see the newly-left room here, or is that checked already in another test?